### DISCUSSION TITLE
Cannot reproduce Flaml predictions using SkLearn RF

### DISCUSSION THREAD

**Discussion by Therrm** [microsoft/FLAML#1054] [Q&A]
Hi there!

After running Flaml on RF only, I get the following best parameters:

`best_hyperparams={"subsample": 1.0, "num_leaves": 256, "n_estimators": 300, "min_split_gain": 0.0, "min_child_samples": 30, "max_depth": -1, "learning_rate": 0.01, "colsample_bytree": 1}`

But when I try to reproduce those predictions with the same parameters using sklearn rf , I get quite different results. For instance, I get only 3 to 4 different predictions while those from Flaml were close to a random distribution.

What else Flaml does that the RF doesn't? Is there some additional post-processing done by Flaml?

Note: I already pre-process my data by removing rows with empty data and normalizing the dataset (for both for Flaml and RF).

Thanks

**Comment by sonichi** [comment]
Is it due to random state? `random_state` is set to 12032022 in flaml.
If this doesn't remove the gap, then it's possible to be due to preprocessing. You can get `feature_transformer` from the AutoML instance. There is no additional post-processing.

**Reply by thinkall** [reply]
+ You can set `skip_transform` to `True` in the `automl_settings`  to avoid feature_transformer.
